shark/iree_utils.py

```diff
@@ -290,18 +290,34 @@ def tensor_to_type_str(input_tensors: tuple, frontend: str):
     """
     list_of_type = []
     for input_tensor in input_tensors:
-        type_string = "x".join([str(dim) for dim in input_tensor.shape])
-        if frontend in ["torch", "pytorch"]:
-            dtype_string = str(input_tensor.dtype).replace("torch.", "")
-        elif frontend in ["tensorflow", "tf"]:
-            dtype = input_tensor.dtype
-            dtype_string = re.findall('\'[^"]*\'',
-                                      str(dtype))[0].replace("\'", "")
-        regex_split = re.compile("([a-zA-Z]+)([0-9]+)")
-        match = regex_split.match(dtype_string)
-        mlir_type_string = str(match.group(1)[0]) + str(match.group(2))
-        type_string += f"x{mlir_type_string}"
-        list_of_type.append(type_string)
+        if (isinstance(input_tensor, tuple)):
+            for val in input_tensor:
+                type_string = "x".join([str(dim) for dim in val.shape])
```

Comment:
Looks like duplicate code, let's move the original function a utility function perhaps calling it something like get_tensor_mlir_type

Reply:
aside from this lgtm! :)